common/src/api/external/mod.rs

```diff
@@ -71,6 +71,23 @@ pub trait ObjectIdentity {
     fn identity(&self) -> &IdentityMetadata;
 }
 
+/// Exists for types that don't properly implement `ObjectIdentity` but
+/// still need to be paginated by name or id.
+pub trait SimpleIdentity {
+    fn id(&self) -> Uuid;
+    fn name(&self) -> &Name;
+}
+
+impl<T: ObjectIdentity> SimpleIdentity for T {
+    fn id(&self) -> Uuid {
+        self.identity().id
+    }
+
+    fn name(&self) -> &Name {
+        &self.identity().name
+    }
+}
+
```

Comment:
sneaky

Reply:
Yeah, the constraint of implementing `ObjectIdentity` felt a little unnecessary. We have several of these not-really-resource resources and we should still be able to treat as if they were (even when they don't have a resource/asset identity)